### ISSUE TITLE
PyPI source archives `SOURCES.txt` includes hardcoded paths

### ISSUE THREAD

**Issue by scdub** [rusty1s/pytorch_sparse#134]
I'm working on building a standalone conda package, and pointing to the source archives hosted in PyPi for the build process. At the end of the build, `setuptools` emits this error, from its parsing of `torch_sparse.egg-info/SOURCES.txt`:

```
  writing torch_sparse.egg-info\PKG-INFO
    writing dependency_links to torch_sparse.egg-info\dependency_links.txt
    writing requirements to torch_sparse.egg-info\requires.txt
    writing top-level names to torch_sparse.egg-info\top_level.txt
    reading manifest file 'torch_sparse.egg-info\SOURCES.txt'
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "C:\Users\conda\AppData\Local\Temp\pip-req-build-502uulsi\setup.py", line 110, in <module>
        packages=find_packages(),
      File "C:\conda\conda-bld\torch-sparse_1619672989686\_h_env\lib\site-packages\setuptools\__init__.py", line 153, in setup
        return distutils.core.setup(**attrs)
      File "C:\conda\conda-bld\torch-sparse_1619672989686\_h_env\lib\distutils\core.py", line 148, in setup
        dist.run_commands()
      File "C:\conda\conda-bld\torch-sparse_1619672989686\_h_env\lib\distutils\dist.py", line 966, in run_commands
        self.run_command(cmd)
      File "C:\conda\conda-bld\torch-sparse_1619672989686\_h_env\lib\distutils\dist.py", line 985, in run_command
        cmd_obj.run()
      File "C:\conda\conda-bld\torch-sparse_1619672989686\_h_env\lib\site-packages\setuptools\command\install.py", line 61, in run
        return orig.install.run(self)
      File "C:\conda\conda-bld\torch-sparse_1619672989686\_h_env\lib\distutils\command\install.py", line 557, in run
        self.run_command(cmd_name)
      File "C:\conda\conda-bld\torch-sparse_1619672989686\_h_env\lib\distutils\cmd.py", line 313, in run_command
        self.distribution.run_command(command)
      File "C:\conda\conda-bld\torch-sparse_1619672989686\_h_env\lib\distutils\dist.py", line 985, in run_command
        cmd_obj.run()
      File "C:\conda\conda-bld\torch-sparse_1619672989686\_h_env\lib\site-packages\setuptools\command\install_egg_info.py", line 34, in run
        self.run_command('egg_info')
      File "C:\conda\conda-bld\torch-sparse_1619672989686\_h_env\lib\distutils\cmd.py", line 313, in run_command
        self.distribution.run_command(command)
      File "C:\conda\conda-bld\torch-sparse_1619672989686\_h_env\lib\distutils\dist.py", line 985, in run_command
        cmd_obj.run()
      File "C:\conda\conda-bld\torch-sparse_1619672989686\_h_env\lib\site-packages\setuptools\command\egg_info.py", line 299, in run
        self.find_sources()
      File "C:\conda\conda-bld\torch-sparse_1619672989686\_h_env\lib\site-packages\setuptools\command\egg_info.py", line 306, in find_sources
        mm.run()
      File "C:\conda\conda-bld\torch-sparse_1619672989686\_h_env\lib\site-packages\setuptools\command\egg_info.py", line 541, in run
        self.add_defaults()
      File "C:\conda\conda-bld\torch-sparse_1619672989686\_h_env\lib\site-packages\setuptools\command\egg_info.py", line 585, in add_defaults
        self.read_manifest()
      File "C:\conda\conda-bld\torch-sparse_1619672989686\_h_env\lib\site-packages\setuptools\command\sdist.py", line 190, in read_manifest
        self.filelist.append(line)
      File "C:\conda\conda-bld\torch-sparse_1619672989686\_h_env\lib\site-packages\setuptools\command\egg_info.py", line 483, in append
        path = convert_path(item)
      File "C:\conda\conda-bld\torch-sparse_1619672989686\_h_env\lib\distutils\util.py", line 110, in convert_path
        raise ValueError("path '%s' cannot be absolute" % pathname)
    ValueError: path '/Users/rusty1s/github/pytorch_sparse/csrc/convert.cpp' cannot be absolute
```

It looks like either `setup.py` or `MANIFEST.in` is causing these files to be included twice, with one copy including the local build path, e.g. `/Users/rusty1s/github/pytorch_sparse/csrc/convert.cpp`, and a second (correct) instance of just the relative `csrc/convert.cpp` path.

**Comment by rusty1s** [comment]
Oh shit, you are absolutely right. Good catch. The bad thing is that I do not yet know why Python adds the absolute paths to `SOURCES.txt`. I need to further dig into this. Any advice appreciated :)

**Comment by rusty1s** [comment]
This fix was resolved easier than it appeared to me at first, see https://github.com/rusty1s/pytorch_scatter/commit/5dce51cef6350b82cf034adb79b4d8f80feb461b. I updated all other external packages accordingly.

**Comment by bwdeng20** [comment]
Sounds great!

> This issue is now resolved, see [rusty1s/pytorch_scatter@5dce51c](https://github.com/rusty1s/pytorch_scatter/commit/5dce51cef6350b82cf034adb79b4d8f80feb461b). I updated all other external packages accordingly.
